src/main/scala/mesosphere/marathon/api/HttpModule.scala

```diff
@@ -0,0 +1,193 @@
+      configureConnectorAddress(connector, conf.httpAddress, conf.httpPort)
+      Some(connector)
+    } else {
+      None
+    }
+  }
+
+  private def getHTTPSConnector(server: Server, httpConfig: HttpConfiguration): Option[ServerConnector] = {
+    def createHTTPSConnector(keystorePath: String, keystorePassword: String): ServerConnector = {
+      val keystore = new File(keystorePath)
+      require(
+        keystore.exists() && keystore.canRead,
+        f"${conf.sslKeystorePath()} is invalid or not readable!")
+
+      val contextFactory = new SslContextFactory()
+      contextFactory.setKeyStorePath(keystorePath)
+      contextFactory.setKeyStorePassword(keystorePassword)
+
+      val sslConfig = new HttpConfiguration(httpConfig)
+      sslConfig.addCustomizer(new SecureRequestCustomizer())
+
+      val sslConnector = new ServerConnector(server, new SslConnectionFactory(contextFactory, HttpVersion.HTTP_1_1.asString()), new HttpConnectionFactory(sslConfig))
+      configureConnectorAddress(sslConnector, conf.httpsAddress, conf.httpsPort)
+
+      sslConnector
+    }
+
+    for {
+      keystorePath <- conf.sslKeystorePath.get
+      keystorePassword <- conf.sslKeystorePassword.get
+      connector = createHTTPSConnector(keystorePath, keystorePassword)
+    } yield connector
+  }
+
+  private[this] def configureConnectorAddress(connector: ServerConnector, addressOpt: ScallopOption[String], portOpt: ScallopOption[Int]): Unit = {
+    connector.setIdleTimeout(30000)
+    addressOpt.foreach(connector.setHost)
+    portOpt.get match {
+      case Some(port) =>
+        connector.setPort(port)
+      case None =>
+        // shouldn't happen because our port configurations all have defaults
+        throw new IllegalArgumentException("Port required.")
+    }
+  }
+
+  lazy val requestLogHandler: RequestLogHandler = {
+    val handler = new RequestLogHandler()
+    handler.setRequestLog(requestLog)
+    handler
+  }
+
+  lazy val handler: ServletContextHandler = {
+    val handler = new ServletContextHandler()
+    conf.httpCredentials.get flatMap createSecurityHandler foreach handler.setSecurityHandler
+    handler
+  }
+
+  lazy val handlerCollection: HandlerCollection = {
+    val c = new HandlerCollection()
+    c.addHandler(handler)
+    c
+  }
+
+  private def createSecurityHandler(httpCredentials: String): Option[ConstraintSecurityHandler] = {
+
+    val credentialsPattern = "(.+):(.+)".r
+
+    httpCredentials match {
+      case credentialsPattern(userName, password) =>
+        Option(createSecurityHandler(userName, password))
+      case _ =>
+        logger.error("The HTTP credentials must be specified in the form of 'user:password'.")
+        None
+    }
+  }
+
+  private def createSecurityHandler(userName: String, password: String): ConstraintSecurityHandler = {
+
+    val constraint = new Constraint(Constraint.__BASIC_AUTH, "user")
+    constraint.setAuthenticate(true)
+
+    //TODO(FL): Make configurable
```

Comment:
Wow. That must be an old todo. Should we remove this?

Reply:
Rofl. Yeah, will do.